RCSwitch.cpp

```diff
@@ -76,7 +79,8 @@ void RCSwitch::setProtocol(Protocol protocol) {
   */
 void RCSwitch::setProtocol(int nProtocol) {
   if (nProtocol < 1 || nProtocol > numProto) {
-    nProtocol = 1;  // TODO: trigger an error, e.g. "bad protocol" ???
+    printf("Warning | RCSwitch::setProtocol | Protocol must be between 1 and %i. Your choice was: %i\n", numProto, nProtocol);
+    nProtocol = 1;
```

Comment:
Sorry, but a printf like this is not acceptable: It increases the code size too much.

It might make sense to change setProtocol to return something to indicate whether an error occurred.